python/ray/serve/tests/test_http_headers.py

```diff
@@ -24,50 +24,52 @@ def __call__(self):
     assert resp.text == resp.headers[RAY_SERVE_REQUEST_ID_HEADER]
 
 
-@pytest.mark.parametrize("deploy_type", ["basic", "fastapi", "starlette_resp"])
-def test_user_provided_request_id_header(ray_shutdown, deploy_type):
-    """Test that a user-provided request_id is propagated to the
-    replica and returned as a header."""
-
-    if deploy_type == "fastapi":
-        app = FastAPI()
+class TestUserProvidedRequestIDHeader:
```

Comment:
No change in test logic, just split up the test within a class and use the serve instance